core/capabilities/test_commons.go

```diff
@@ -0,0 +1,115 @@
+package capabilities
+
+import (
+	"crypto/rand"
+
+	ragetypes "github.com/smartcontractkit/libocr/ragep2p/types"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities"
+	kcr "github.com/smartcontractkit/chainlink-evm/gethwrappers/keystone/generated/capabilities_registry_1_1_0"
+	p2ptypes "github.com/smartcontractkit/chainlink/v2/core/services/p2p/types"
+	"github.com/smartcontractkit/chainlink/v2/core/services/registrysyncer"
+)
+
+func randomWord() [32]byte {
+	word := make([]byte, 32)
+	_, err := rand.Read(word)
+	if err != nil {
+		panic(err)
+	}
+	return [32]byte(word)
+}
+
+type TestTopology struct {
+	workflowDonNodes   []p2ptypes.PeerID
+	capabilityDonNodes []p2ptypes.PeerID
+}
+
+func (tt *TestTopology) MakeNodes(count int) []p2ptypes.PeerID {
+	nodes := make([]p2ptypes.PeerID, count)
+	for i := range nodes {
+		nodes[i] = randomWord()
+	}
+	return nodes
+}
+
+func (tt *TestTopology) DonMaker(dID uint32, donNodes []p2ptypes.PeerID, acceptWorkflow bool) capabilities.DON {
```

Comment:
I wonder if similar things already exist in other places. It's fine to keep it but if you ever discover an overlap then please consolidate.